### PR TITLE
[feat]: [CDS-76134]: Support for consoleLogPollFrequency jenkins build

### PR DESCRIPTION
**Pull request by Abhishek-CDC**
## Describe your changes
Support a new field 'consoleLogPollFrequency' to allow user to manage poll task frequency.
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51622)

### CODE REVIEW

**Comment by Abhishek-CDC** [comment]
trigger uts

**Review by shivamnegi94 — DISMISSED** [review]

File: 125-cd-nextgen/src/main/java/io/harness/cdng/jenkins/jenkinsstep/JenkinsBuildStepHelperServiceImpl.java [hunk 1]
```diff
@@ -92,7 +93,7 @@ public class JenkinsBuildStepHelperServiceImpl implements JenkinsBuildStepHelper
   public JenkinsBuildStepHelperServiceImpl(ConnectorResourceClient connectorResourceClient,
       @Named("PRIVILEGED") SecretManagerClientService secretManagerClientService,
       @Named("referenceFalseKryoSerializer") KryoSerializer kryoSerializer,
-      LogStreamingStepClientFactory logStreamingStepClientFactory) {
+      LogStreamingStepClientFactory logStreamingStepClientFactory, NGSettingsClient ngSettingsClient) {
```
Comment (puthrayaharness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/jenkins/jenkinsstep/JenkinsBuildStepHelperServiceImpl.java:96]: Remove unwanted change

File: 125-cd-nextgen/src/main/java/io/harness/cdng/jenkins/jenkinsstep/JenkinsBuildStepHelper.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
+@UtilityClass
+@Slf4j
+public class JenkinsBuildStepHelper {
+  static final String DEFAULT_CONSOLE_LOG_FREQUENCY = "5s";
+  static final long DEFAULT_CONSOLE_LOG_FREQUENCY_SECONDS = 5;
+  private final String CONSOLE_LOG_FREQUENCY = "Jenkins step consoleLogPollFrequency";
+  private final String ERROR_MESSAGE = "Valid value contain units s/m/h/d/w and are written like 10s or 1m 10s";
+
+  long getConsoleLogPollingFrequency(ParameterField<String> frequency) {
```
Comment (puthrayaharness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/jenkins/jenkinsstep/JenkinsBuildStepHelper.java:31]: Please log these to the HarnessUI console, not just to our app logs.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/jenkins/jenkinsstep/JenkinsBuildStepHelper.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    long result = DEFAULT_CONSOLE_LOG_FREQUENCY_SECONDS;
```
Comment (puthrayaharness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/jenkins/jenkinsstep/JenkinsBuildStepHelper.java:32]: Please avoid so much nesting in code. Consider returning early to make the code less nested & in turn makes the code a lot more readable.

Comment (puthrayaharness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/jenkins/jenkinsstep/JenkinsBuildStepHelper.java:32]: We can get on a call & resolve this.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Comment by Abhishek-CDC** [comment]
trigger uts

**Review by puthrayaharness — APPROVED** [review]

**Review by shivamnegi94 — APPROVED** [review]

**Comment by Abhishek-CDC** [comment]
trigger review

**Pull request merged**